src/components/button/_mixins.scss

```diff
@@ -153,6 +142,25 @@
     border: $button-outline-width solid transparent;
   }
 
+  &:disabled:hover,
+  &:disabled:focus {
+    text-decoration: none;
+  }
+}
```

Comment:
I would expect the `button-theme` mixin to call the `button-theme-color` mixin like in accordion. https://github.com/IBM/carbon-components/blob/9504fa4d6a3470275cd9eb6f5f5f2d3e6a503f4c/src/components/accordion/_accordion.scss#L277

Another question about future expectations: would `button-theme` become `button` and `button-theme-color` become `button-color` to align with what is going on with accordion?

Reply:
There is no accordion correspondent of `@include button-theme()`. Though I'm not the one who created it, I see it's for allowing different button variants. I see similar patterns in other design systems:

```scss
.#{$prefix}--btn--primary {
  @include button-theme($brand-01, transparent, $inverse-01, $hover-primary, $ui-01);
}

.#{$prefix}--btn--secondary {
  @include button-theme(transparent, $brand-01, $brand-01, $hover-secondary, $brand-01);
  ...
}

.#{$prefix}--btn--tertiary {
  @include button-theme(transparent, $ui-05, $ui-05, $ui-05, $ui-05);
  ...
}
```

Reply:
I don't know where the comment went but I swear I saw an email with a response to this.  To be clear, for buttons the `button-theme` mixin is actually a `variant`. The name `button-theme` is confusing given the new inline-theming/zone option that this PR introduces.  I wonder if it would be smart to take advantage of the Carbon X release and rename this to be clearer to something like `button-variant` instead.